### ISSUE TITLE
Default Custom Controls Still Not Being Set

### ISSUE THREAD

**Issue by MjnMixael** [scp-fs2open/fs2open.github.com#3792]
As seen during RedMageJoe's recent BtA stream, the custom controls we have defined are not being applied as defaults for new players. Doubly so, they aren't being saved across campaigns with the same player within the same mod. They have to manually bind them each time.

RedMageJoe, here notices the key is not bound mid-mission and binds them.
https://youtu.be/Nv5yu1e8F7A?t=5673
Later in the same stream when he moves to the next campaign in BtA, the controls are again not bound and have to be bound again.

Ideally, players starting the campaign should have these keybinds automatically applied.

Here is the contents of the BtA controlconfigdefaults.tbl

`#ControlConfigOverride

$Bind Name: Custom Control 1
$Key Default: KEY_A
$Key Mod Shift: 1
$Key Mod Alt: 1
$Category: SHIP_TAB
$Has XStr: 1639
$Disable: false

$Bind Name: Custom Control 2
$Key Default: KEY_S
$Key Mod Shift: 1
$Key Mod Alt: 1
$Category: SHIP_TAB
$Has XStr: 1640
$Disable: false

$Bind Name: Custom Control 3
$Key Default: KEY_D
$Key Mod Shift: 1
$Key Mod Alt: 1
$Category: SHIP_TAB
$Has XStr: 1641
$Disable: false

$Bind Name: Custom Control 4
$Key Default: KEY_Z
$Key Mod Shift: 1
$Key Mod Alt: 1
$Category: SHIP_TAB
$Has XStr: 1642
$Disable: false

$Bind Name: Custom Control 5
$Key Default: KEY_X
$Key Mod Shift: 1
$Key Mod Alt: 1
$Category: SHIP_TAB
$Has XStr: 1643
$Disable: false

#End`

**Comment by EatThePath** [comment]
I decided to sniff at this in a debugger and I wasn't able to entirely reproduce. RMJ is as far as I know on 21.4.0 stable, so I used that, and made a new pilot on retail without touching the binds. Then I fired up BTA and got in game. All the special keys are bound at that point, though to the wrong things. They're bound to alt+shift+1 through 5. Then mid-mission I bound remote detonate to = and switched campaigns and jumped back in, and all the binds were unchanged by the switch. Without much familiarity with controls code, I'm not sure where to start looking for the missing factor.

**Comment by z64555** [comment]
Need to try repro'ing this with the Controls5 changes during RC.

**Comment by jg18** [comment]
I can't repro this issue with latest BtA release (1.6.11) and FSO 21.4.0 (mod default) on Windows 10.

I created a new pilot and checked the controls menu. The mod's custom controls' default key bindings are there.

I then changed one of the custom controls' bindings and clicked on "Defaults". The binding changes back to the mod's default.

I then switched to the next campaign and tried the same test. Everything worked fine.

Closing this issue until someone can reliably repro it.
